src/server/chromium/chromium.ts

```diff
@@ -128,7 +131,77 @@ export class Chromium extends BrowserType {
     transport.send(message);
   }
 
+  override async _launchWithSeleniumHub(progress: Progress, hubUrl: string, options: types.LaunchOptions): Promise<CRBrowser> {
+    if (!hubUrl.endsWith('/'))
+      hubUrl = hubUrl + '/';
+
+    const args = this._innerDefaultArgs(options);
+    args.push('--remote-debugging-port=0');
+    const desiredCapabilities = { 'browserName': 'chrome', 'goog:chromeOptions': { args } };
+
+    progress.log(`<connecting to selenium> ${hubUrl}`);
+    const response = await fetchData({
+      url: hubUrl + 'session',
+      method: 'POST',
+      data: JSON.stringify({
+        desiredCapabilities,
+        capabilities: { alwaysMatch: desiredCapabilities }
+      }),
+      timeout: progress.timeUntilDeadline(),
+    }, async response => {
+      const body = await streamToString(response);
+      let message = '';
+      try {
+        const json = JSON.parse(body);
+        message = json.value.localizedMessage || json.value.message;
+      } catch (e) {
+      }
+      return new Error(`Error connecting to Selenium at ${hubUrl}: ${message}`);
+    });
+    const value = JSON.parse(response).value;
+    const sessionId = value.sessionId;
+    progress.log(`<connected to selenium> sessionId=${sessionId}`);
+
+    const disconnectFromSelenium = async () => {
+      progress.log(`<disconnecting from selenium> sessionId=${sessionId}`);
+      await fetchData({
+        url: hubUrl + 'session/' + sessionId,
+        method: 'DELETE',
+      }).catch(() => {});
+      progress.log(`<disconnected from selenium> sessionId=${sessionId}`);
+    };
+
+    try {
+      const capabilities = value.capabilities;
+      const maybeChromeOptions = capabilities['goog:chromeOptions'];
+      const chromeOptions = maybeChromeOptions && typeof maybeChromeOptions === 'object' ? maybeChromeOptions : undefined;
+      const debuggerAddress = chromeOptions && typeof chromeOptions.debuggerAddress === 'string' ? chromeOptions.debuggerAddress : undefined;
+      const chromeOptionsURL = typeof maybeChromeOptions === 'string' ? maybeChromeOptions : undefined;
+      let endpointURL = capabilities['se:cdp'] || debuggerAddress || chromeOptionsURL;
+      if (!['ws://', 'wss://', 'http://', 'https://'].some(protocol => endpointURL.startsWith(protocol)))
+        endpointURL = 'http://' + endpointURL;
```

Comment:
log this `endpointURL` too?

Reply:
`connectOverCDP` does it for us already.